framework-plugins/lisk-framework-forger-plugin/src/forger_plugin.ts

```diff
@@ -114,27 +122,57 @@ export class ForgerPlugin extends BasePlugin {
 		const options = objects.mergeDeep({}, config.defaultConfig.default, this.options) as Options;
 		this._channel = channel;
 
+		// eslint-disable-next-line new-cap
+		const { locale } = Intl.DateTimeFormat().resolvedOptions();
+
+		this._webhooks = new Webhooks(
+			{
+				'User-Agent': `lisk-framework-forger-plugin/0.1.0 (${os.platform()} ${os.release()}; ${os.arch()} ${locale}.${
+					process.env.LC_CTYPE ?? ''
+				}) lisk-framework/${options.version}`,
+			},
+			options.webhook,
+		);
 		this._forgerPluginDB = await getDBInstance(options.dataPath);
 
 		// eslint-disable-next-line @typescript-eslint/no-misused-promises
 		this._channel.once('app:ready', async () => {
 			this._app = initApi(options, this._channel, this.codec, this._forgerPluginDB);
 
+			// eslint-disable-next-line @typescript-eslint/no-floating-promises
+			this._webhooks.handleEvent({
+				event: 'forger:node:start',
+				time: Date.now(),
+				payload: { reason: 'Node started' },
+			});
+
 			// Fetch and set forger list from the app
 			await this._setForgersList();
 
 			// Fetch and set transactions fees
 			await this._setTransactionFees();
 
 			// Sync the information
+			this._syncingWithNode = true;
 			await this._syncForgerInfo();
+			this._syncingWithNode = false;
 
 			// Listen to new block and delete block events
 			this._subscribeToChannel();
 
 			// Start http server
 			this._server = this._app.listen(options.port, '0.0.0.0');
 		});
+
+		// @TODO Fix me! due to the way unload works this event is never fired in time.
```

Comment:
is this related to what @shuse2 is working on, if so maybe link his issue number here.. so that TODO can be fixed

Reply:
Not sure if related. The problem I think it's related to the fact that unload does not have a handling of any potential pending events on any of the subscribers. For this to work (and report the actual node stopping instead of the plugin being unloaded) we should be able to implement some blocking mechanism based on any pending received event I think. But maybe we can discuss about later.